src/bun.js/api/bun/socket.zig

```diff
@@ -2114,11 +2113,17 @@ fn NewSocket(comptime ssl: bool) type {
                 .onError = this.handlers.onError,
                 .onHandshake = this.handlers.onHandshake,
                 .binary_type = this.handlers.binary_type,
+                .is_server = this.handlers.is_server,
             };
-
-            raw_handlers_ptr.* = cloned_handlers;
-            raw_handlers_ptr.is_server = this.handlers.is_server;
-            raw_handlers_ptr.protect();
+            this.handlers.onOpen = .zero;
+            this.handlers.onClose = .zero;
```

Comment:
is the idea here that we detach the previous handlers?

Reply:
yes